core/internal/cluster/kafka_cluster.go

```diff
@@ -278,3 +299,38 @@ func (module *KafkaCluster) getOffsets(client helpers.SaramaClient) {
 		return false
 	})
 }
+
+func (module *KafkaCluster) reapNonExistingGroups(client helpers.SaramaClient) {
+	kafkaGroups, err := client.ListConsumerGroups()
+	if err != nil {
+		module.Log.Warn("failed to get the list of available consumer groups", zap.Error(err))
+	}
+
+	req := &protocol.StorageRequest{
+		RequestType: protocol.StorageFetchConsumers,
+		Reply:       make(chan interface{}),
+		Cluster:     module.name,
+	}
+	helpers.TimeoutSendStorageRequest(module.App.StorageChannel, req, 20)
+
+	res := <-req.Reply
+	if res == nil {
+		module.Log.Warn("groups reaper: couldn't get list of consumer groups from storage")
+		return
+	}
+
+	// TODO: find how to get reportedConsumerGroup from KafkaClient
+	burrowIgnoreGroupName := "burrow-" + module.name
+	burrowGroups, _ := res.([]string)
+	for _, g := range burrowGroups {
+		if _, ok := kafkaGroups[g]; !ok && g != burrowIgnoreGroupName {
+			module.Log.Info(fmt.Sprintf("groups reaper: removing non existing kafka consumer group (%s) from burrow", g))
+			request := &protocol.StorageRequest{
+				RequestType: protocol.StorageSetDeleteGroup,
+				Cluster:     module.name,
+				Group:       g,
+			}
+			helpers.TimeoutSendStorageRequest(module.App.StorageChannel, request, 1)
```

Comment:
Is it processed in another routine?

Reply:
yes, it'll eventually be processed by the storage channel reader